test/language/class/class-name-binding-in-extends-expression.js

```diff
@@ -0,0 +1,18 @@
+// Copyright (C) Copyright 2014 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+es6id: 14.5
+description: >
+    class name binding in extends expression
+---*/
+assert.throws(ReferenceError, function() {
+  class x extends x {}
+});
+
+assert.throws(ReferenceError, function() {
+  (class x extends x {});
+});
+
+assert.throws(ReferenceError, function() {
+  var x = (class x extends x {});
```

Comment:
Per https://people.mozilla.org/~jorendorff/es6-draft.html#sec-runtime-semantics-classdefinitionevaluation 6.f this should be a TypeError not a reference error. X is declared due to var hoisting, but its value is undefined failing the IsConstructor check.